app/workers/notification_worker.rb

```diff
@@ -64,4 +64,13 @@ def lists_matched_on_links
     @lists_matched_on_links ||= SubscriberListQuery.new(query_field: :links)
       .where_all_links_match_at_least_one_value_in(@links_hash)
   end
+
+  def filter_by_document_type(matching_lists)
+    matching_lists.select { |l| l.document_type.blank? || l.document_type == @document_type }
+  end
+
+  def lists
+    matching_lists = (lists_matched_on_links + lists_matched_on_tags).uniq { |l| l.id }
+    filter_by_document_type(matching_lists)
+  end
```

Comment:
Is this kind of filtering what the `SubscriberListQuery` is for, or is that for something else?